processing/src/main/java/org/apache/druid/segment/AggregateProjectionMetadata.java

```diff
@@ -182,9 +182,15 @@ public Schema(
         @JsonProperty("ordering") List<OrderBy> ordering
     )
     {
+      if (name == null) {
+        throw DruidException.defensive("projection schema name cannot be null");
```

Comment:
Should these be defensive or invalid input?

Reply:
these are defensive because this object is created internally from `AggregateProjectionSpec` which is the user facing object that does invalid input